src/backend/utils/adt/selfuncs.c

```diff
@@ -1241,6 +1241,8 @@ patternsel(PG_FUNCTION_ARGS, Pattern_Type ptype, bool negate)
 
 		if (eqopr == InvalidOid)
 			elog(ERROR, "no = operator for opfamily %u", opfamily);
+
+		Insist(prefix != NULL);
```

Comment:
https://github.com/arenadata/gpdb/blob/cbef1396b38e20a82ba358b157445e4d673fc7ff/src/backend/utils/adt/selfuncs.c#L1205-L1206

Maybe we should just do this right under the `pattern_fixed_prefix()` call:
```c
		AssertImply(pstatus != Pattern_Prefix_None, prefix != NULL);
```
?

Reply:
Yes, but this will not save us from the static analyzer.

Reply:
Can we get rid of the warning with this instead of each `Insist()`?
```c
if (pstatus != Pattern_Prefix_None)
	Insist(prefix != NULL);
```

Reply:
I think, no, because [condition](https://github.com/arenadata/gpdb/pull/770/files#diff-60407b5c46aaf9ea36ed1002b1bfaf5e818298c30d3d680c4e9c91f814befd46R1211) was left